### PR TITLE
Add send-email function

### PR DESCRIPTION
**Pull request by keirog**
Expects the following object in the event:
```
const emailConfig = {
	userUuid: 'abc123',
	htmlContent: '<html><p>HTML content</p></html>',
	plainTextContent: 'Plain text',
	emailType: 'daily'
};
```
Still to do:
- [ ] Implement retries on non-410 errors
- [ ] Implement sending to queues (unsubscribe-queue for 410s, retry-queue for other errors)

### CODE REVIEW

File: functions/send-email/lib/catch-error.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = () => {};
```
Comment (phamann) [inline functions/send-email/lib/catch-error.js]: Is this intentionally empty? Would be good to track errors via `metric.count` and `logger.error`

File: functions/send-email/lib/send-to-email-service.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	},
+	body: {
+		transmissionHeader: {
+			metadata: {}
+		},
+		from: {
+			address: 'myft@news-alerts.ft.com',
+			name: 'myFT'
+		},
+		subject: 'myFT Digest'
```
Comment (phamann) [inline functions/send-email/lib/send-to-email-service.js]: Can we add digest type here too: "myFT Daily Digest"?

File: functions/send-email/lib/send-to-email-service.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	const requestOptions = Object.assign({}, defaults, { body });
+	requestOptions.body = JSON.stringify(requestOptions.body);
+	return requestOptions;
+}
+
+const validateConfig = config => {
+	const receivedKeys = Object.keys(config).sort().join(',');
+	const expectedKeys = ['emailType', 'htmlContent', 'plainTextContent', 'userUuid'].join(',');
+	return receivedKeys === expectedKeys;
+}
```
Comment (phamann) [inline functions/send-email/lib/send-to-email-service.js]: Nice 😎

File: functions/send-email/lib/send-to-email-service.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+module.exports = config => {
+	return new Promise((resolve, reject) => {
```
Comment (phamann) [inline functions/send-email/lib/send-to-email-service.js]: You don't need to construct a new promise here.

```
if (!validateConfig(config)) return Promise.reject(error)
return fetch(etc)
```

**Comment by phamann** [comment]
👍 Other than comments

File: functions/send-email/lib/catch-error.js (hunk 1 above)
Comment (keirog) [inline functions/send-email/lib/catch-error.js]: Ah yep left empty for now; it'll be where the 4**/5** responses get sent to the appropriate queue (unsubscribe & retry queues)

File: functions/send-email/lib/send-to-email-service.js (hunk 2 above)
Comment (keirog) [inline functions/send-email/lib/send-to-email-service.js]: I thought we'd agreed in the meeting yesterday to drop the Daily/Weekly bit in the subject, but maybe not. Will check with James

File: functions/send-email/lib/send-to-email-service.js (hunk 4 above)
Comment (keirog) [inline functions/send-email/lib/send-to-email-service.js]: Ah yep, thanks!

**Pull request merged**